src/main/java/io/github/wimdeblauwe/errorhandlingspringbootstarter/mapper/ErrorCodeMapper.java

```diff
@@ -18,16 +18,12 @@ public ErrorCodeMapper(ErrorHandlingProperties properties) {
     }
 
     public String getErrorCode(Throwable exception) {
-        String exceptionClassName = exception.getClass().getName();
-        if (properties.getCodes().containsKey(exceptionClassName)) {
-            return properties.getCodes().get(exceptionClassName);
-        }
-
-        ResponseErrorCode errorCodeAnnotation = AnnotationUtils.getAnnotation(exception.getClass(), ResponseErrorCode.class);
-        if (errorCodeAnnotation != null) {
-            return errorCodeAnnotation.value();
+         // Find the first existing error code override throw the class hierarchy
+         String code = getErrorCode(exception.getClass());
```

Comment:
The comments are not needed. I would rename the private `getErrorCode` method to `getErrorCodeFromPropertiesOrAnnotation`

Reply:
Ok. I'll remove all inline comments (thought those could be useful) and change the method names.